### ISSUE TITLE
Show number of invocations along with the time in the frontend

### ISSUE THREAD

**Issue by camba1** [grafana/pyroscope#59]
The frontend should show the total number of invocations of a function along with its total time in the charts. It is good to know that a function is taking 4s because it was called 20 times vs another function that was called 2 times and took 3s.

**Comment by petethepig** [comment]
This is tricky because the profilers we use are all sampling profilers — they look at the current stacktrace of a given program X times per second. So there's no information about how many times a certain function got called. Good thing about this is that they are significantly less taxing in terms of CPU overhead. The bad thing is... well, we can't get number of invocations for example :)

We could at some point add support for other types of profilers and this could become possible.

Also, I kinda wonder exactly how this information can be helpful. In my experience when I'm optimizing some program I usually look at the current slowest function and I optimize that one until it's no longer the slowest one, and then I repeat the process until I'm happy with the results or get diminishing returns. And so in this case it usually doesn't matter how many times a function gets called — as long as I know that one of them takes up more time that's enough information for me to start digging in a particular direction.

I guess I could see this being useful when the number of calls is significantly lower / higher than what you would expect — that fact alone could lead to some conclusions about the potential optimizations.

@camba1 Is there any other situations when this information can be helpful? Could you share some specific examples? It would be super helpful for us to understand the use-case better.

Thank you!

**Comment by camba1** [comment]
Hum... I see how that would be a hard number with sampling profilers. So maybe this is a feature to be added later if you decide to use other profilers or if you allow user specified sampling rate. The user could set it very high and get a potentially meaningful number. Anyways, this may be a long term thing.

In terms of situations when this would useful. Here is a very simplified real world example:
We were calculating sales KPIs for a business unit that had 500 products. We noticed that over the duration of the process function X was taking a total of 5s (I do not recall the exact numbers, so using directional correct numbers ). We also noticed that function Y was taking 1s. By just looking at those numbers one would think that function X was higher in priority for optimization. However, that was just part of the story because  function X was being called in the calculation of each of the 500 product's KPIs while function Y was only applicable to the KPI's of 5 products
In other words, function X was taking in average .01s to run (5/500) while function Y was taking in average of .5s (1/2).   

Another useful use of the number of invocations is monitoring the average duration per invocation over time. If the average duration grows without the number of number of invocations growing as well, then you could have something  causing the performance to degrade

Hope that helps.

**Comment by petethepig** [comment]
@camba1 Thank you! This is exactly what I was looking for, very helpful.

I think this would require a different type of profiler, so definitely more of a long-term thing.

**Comment by petethepig** [comment]
Somewhat similar feedback https://news.ycombinator.com/reply?id=26486923
